aten/src/ATen/native/native_functions.yaml

```diff
@@ -1815,7 +1821,7 @@
 - func: diag_embed(Tensor self, int offset=0, int dim1=-2, int dim2=-1) -> Tensor
   variants: function, method
   dispatch:
-    CompositeExplicitAutogradNonFunctional: diag_embed
+    CompositeExplicitAutograd: diag_embed
```

Comment:
hmmm, why?

`CompositExplicitAutogradNonFunctional` is a bit funny today - it's basically only used for providing better error messages for XLA/LTC. If we have a functional op that calls a view op, that is `CompositeExplicitAutograd`, then XLA and LTC can't handle it directly so we don't want to register it to them. Not clear to me why we had to remove it for `diag_embed` here though.

Reply:
Not sure tbh, I haven't reviewed this. Reverting.